gap/semigroups/semirms.gi

```diff
@@ -881,16 +881,21 @@ function(R)
   return MagmaIsomorphismByFunctionsNC(R, S, iso, inv);
 end);
 
-InstallMethod(IsIdempotentGenerated, "for a Rees 0-matrix semigroup",
-[IsReesZeroMatrixSemigroup],
+InstallMethod(IsIdempotentGenerated, "for a Rees 0-matrix subsemigroup",
+[IsReesZeroMatrixSubsemigroup],
 function(R)
-  local RR;
-  if not IsConnectedDigraph(RZMSDigraph(R)) then
+  local U, N;
+  if not IsReesZeroMatrixSemigroup(R) then
+    TryNextMethod();
+  elif not IsConnectedDigraph(RZMSDigraph(R)) then
     return false;
   fi;
-  RR := Range(RZMSNormalization(R));
-  return Group(Filtered(MatrixEntries(RR), x -> x <> 0)) =
-    UnderlyingSemigroup(RR);
+  U := UnderlyingSemigroup(R);
+  if not IsGroupAsSemigroup(U) then
+    TryNextMethod();
+  fi;
+  N := Range(RZMSNormalization(R));
+  return Semigroup(Filtered(MatrixEntries(N), x -> x <> 0)) = U;
 end);
```

Comment:
Just to double check, since you are replacing a more specific method with a more general one (which is good, of course!) is there any loss of performance in the case that the more specific method applies?